File: Đồ án OOP Java - Quản lý phòng vé bán xe khách/src/Doan2/Chuyenxe.java
package Doan2;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.InputStreamReader;

public abstract class Chuyenxe
{
	protected int iMaso;
	protected String sTentaixe;
	protected String sBienso;
	protected double iDoanhthu;
	protected double iGiave;
	protected int iSove;
	BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
	//
	public int getMaso()
	{
		return this.iMaso;
	}
	public void setMaso(int value)
	{
		this.iMaso = value;
	}
	public String getTentaixe()
	{
		return this.sTentaixe;
	}
	public void setTentaixe(String value)
	{
		this.sTentaixe = value;
	}
	public String getBienso()
	{
		return this.sBienso;
	}
	public void setBienso(String value)
	{
		this.sBienso = value;
	}
	public double getDoanhthu()
	{
		return this.iDoanhthu;
	}
	public void setDoanhthu(Double value)
	{
		this.iDoanhthu = value;
	}
	public double getGiave()
	{
		return this.iGiave;
	}
	public void setGiave(Double value)
	{
		this.iGiave = value;
	}
	public int getSove()
	{
		return this.iSove;
	}
	public void setSove(int value)
	{
		this.iSove = value;
	}
	//
	public Chuyenxe()
	{

	}
	//
	public void Nhap() throws  IOException
	{
		System.out.print("Nhập mã số tuyến : ");
		this.iMaso = Integer.parseInt(input.readLine());
		System.out.print("Nhập tên tài xế: ");
		this.sTentaixe = input.readLine();
		System.out.print("Nhập biển số: ");
		this.sBienso = input.readLine();
		System.out.print("Nhập giá vé: ");
		this.iGiave = Double.parseDouble(input.readLine());
		System.out.print("Nhập số vé: ");
		this.iSove = Integer.parseInt(input.readLine());
	}
	public abstract void Xuat();
	public abstract void GhifileTxt(String filename) throws IOException;
	public abstract void DocfileTxt(String filename) throws FileNotFoundException, IOException;
	public abstract double Tinhdoanhthu();
	public abstract void Them() throws NumberFormatException, IOException;
	public abstract void Sua(int temp) throws IOException;
	public abstract void Xoa(int temp) throws IOException;
	public abstract void Sort() throws IOException;
	public abstract void Search() throws NumberFormatException, IOException;
	public abstract void chuyenmax();
	public abstract void chuyenmin();

}

File: Đồ án OOP C# - Quản lý phòng bán vé xe khách/DoAn2/Chuyenxe.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DoAn2
{
    public abstract class Chuyenxe
    {
        protected int iMaso;
        protected string sTentaixe;
        protected string sBienso;
        protected double iDoanhthu;
        protected double iGiave;
        protected int iSove;
        //
        public int Maso
        {
            get { return this.iMaso; }
            set { this.iMaso = value; }
        }
        public string Tentaixe
        {
            get { return this.sTentaixe; }
            set { this.sTentaixe = value; }
        }
        public string Bienso
        {
            get { return this.sBienso; }
            set { this.sBienso = value; }
        }
        public double Doanhthu
        {
            get { return this.iDoanhthu; }
            set { this.iDoanhthu = value; }
        }
        public double Giave
        {
            get { return this.iGiave; }
            set { this.iGiave = value; }
        }
        public int Sove
        {
            get { return this.iSove; }
            set { this.iSove = value; }
        }
        //
        public Chuyenxe()
        {

        }
        /* public Chuyenxe(int Maso, string Tentaixe, string Bienso, int Doanhthu, int Sovedaban, int Giave)
         {
             this.iMaso = Maso;
             this.sTentaixe = Tentaixe;
             this.sBienso = Bienso;
             this.iDoanhthu = Doanhthu;
             this.iSovedaban = Sovedaban;
             this.iGiave = Giave;
         }
         public Chuyenxe(int Maso, string Tentaixe, string Bienso, int Doanhthu, int Sovedaban)
         {
             this.iMaso = Maso;
             this.sTentaixe = Tentaixe;
             this.sBienso = Bienso;
             this.iDoanhthu = Doanhthu;
             this.iSovedaban = Sovedaban;
         }
         public Chuyenxe(int Maso, string Tentaixe, string Bienso, int Doanhthu)
         {
             this.iMaso = Maso;
             this.sTentaixe = Tentaixe;
             this.sBienso = Bienso;
             this.iDoanhthu = Doanhthu;
         }
         public Chuyenxe(int Maso, string Tentaixe, string Bienso)
         {
             this.iMaso = Maso;
             this.sTentaixe = Tentaixe;
             this.sBienso = Bienso;
         }
         public Chuyenxe(int Maso, string Tentaixe)
         {
             this.iMaso = Maso;
             this.sTentaixe = Tentaixe;
         }
         public Chuyenxe(int Maso)
         {
             this.iMaso = Maso;
         }*/
        //
        public void Nhap()
        {
            Console.Write("Nhập mã số tuyến : ");
            this.iMaso = int.Parse(Console.ReadLine());
            Console.Write("Nhập tên tài xế: ");
            this.sTentaixe = Console.ReadLine();
            Console.Write("Nhập biển số: ");
            this.sBienso = Console.ReadLine();
            Console.Write("Nhập giá vé: ");
            this.iGiave = double.Parse(Console.ReadLine());
            Console.Write("Nhập số vé: ");
            this.iSove = int.Parse(Console.ReadLine());
            // Console.Write("Nhập doanh thu: ");
            //this.iDoanhthu = Convert.ToInt32(Console.ReadLine());
        }
        public abstract void Xuat();
        public abstract void GhifileTxt(string filename);
        public abstract void DocfileTxt(string filename);
        public abstract double Tinhdoanhthu();
        public abstract void Them();
        public abstract void Sua(int temp);
        public abstract void Xoa(int temp);
        public abstract void Sort();
        public abstract void Search();
        public abstract void chuyenmax();
        public abstract void chuyenmin();

    }
}
